builder/proxmox/common/step_finalize_template_config.go

```diff
@@ -60,10 +60,27 @@ func (s *stepFinalizeTemplateConfig) Run(ctx context.Context, state multistep.St
 			}
 		}
 		if cloudInitStoragePool != "" {
-			ideControllers := []string{"ide0", "ide1", "ide2", "ide3"}
+			var diskControllers []string
+			switch c.CloudInitDiskType {
+			// Proxmox supports up to 6 SATA controllers (0 - 5)
+			case "sata":
+				for i := 0; i < 6; i++ {
+					sataController := fmt.Sprintf("sata%d", i)
+					diskControllers = append(diskControllers, sataController)
+				}
+			// and up to 31 SCSI controllers (0 - 30)
+			case "scsi":
+				for i := 0; i < 31; i++ {
+					scsiController := fmt.Sprintf("scsi%d", i)
+					diskControllers = append(diskControllers, scsiController)
+				}
+			default:
```

Comment:
I would suggest defaulting to ide only if it's empty or explicitly `ide`, otherwise this allows for anything and I can see this being a surprise potentially.

```suggestion
			"", "ide":
```

Reply:
EDIT: nevermind I notice we're in `Run` here, I'd suggest then having that check/error in the `Prepare` function then, so we can set `ide` if unset for example, and we can error if the option has an unsupported value.

Reply:
I have added a default, check, and error in the `Prepare` function. Does this look good?
I tested it with a `CloudInitDiskType` value that was not supported (not `scsi`,`sata`, or `ide`) and it stopped packer with the error message I added. I also confirmed that blanks still default to `ide`.
https://github.com/hashicorp/packer-plugin-proxmox/pull/256/commits/e03831f1da79eed6ff899fff8d4ad9258d0087ff

Reply:
I would still suggest expliciting `ide` for this case, just in case something else manages to go through the `Prepare`, so that it doesn't go into `ide` mode when reaching this step.
Adding a `default` with an error is probably for the best for later developments, as this will error in case we update `Prepare` but forget to do the same here.
```suggestion
			"ide":
				// Unspecified disk type defaults to "ide"
				diskControllers = []string{"ide0", "ide1", "ide2", "ide3"}
			default:
				state.Put("error", fmt.Errorf("unsupported disk type %q", c.CloudInitDiskType))
				return multistep.ActionHalt
```